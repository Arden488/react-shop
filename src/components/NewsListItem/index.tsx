import * as PropTypes from 'prop-types';
import * as React from 'react';
import { Link } from 'react-router-dom';

interface INewListItemProps {
  code: string,
  title: string
}

const NewsListItem: React.SFC<INewListItemProps> = props => (
  <div>
    <Link to={`/news/${props.code}`}>{props.title}</Link>
  </div>
);

NewsListItem.propTypes = {
  code: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default NewsListItem;
