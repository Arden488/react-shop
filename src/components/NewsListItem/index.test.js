/* global it, expect, beforeEach */

import React from 'react';
import { shallow } from 'enzyme';
import { Link } from 'react-router-dom';

import NewsListItem from './index';

let component = null;

beforeEach(() => {
  component = shallow(<NewsListItem code="1" title="1" />);
});

it('renders without crashing', () => {
  expect(component.exists()).toEqual(true);
});

it('should have Link', () => {
  expect(component.find(Link).length).toBeGreaterThanOrEqual(1);
});
