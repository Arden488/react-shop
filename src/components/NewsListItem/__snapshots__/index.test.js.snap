// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <Link
    replace={false}
    to="/news/1"
  >
    1
  </Link>
</div>
`;
